extensions/2.0/Khronos/KHR_meshopt_compression/README.md

```diff
@@ -0,0 +1,675 @@
+	float32_t one = input[3] | 3;
+
+	float32_t x = input[0] / one * range;
+	float32_t y = input[1] / one * range;
+	float32_t z = input[2] / one * range;
+
+	float32_t w = sqrt(max(0.0, 1.0 - x * x - y * y - z * z));
+
+	int maxcomp = input[3] & 3;
+
+	// maxcomp specifies a cyclic rotation of the quaternion components
+	output[(maxcomp + 1) % 4] = round(x * 32767.0);
+	output[(maxcomp + 2) % 4] = round(y * 32767.0);
+	output[(maxcomp + 3) % 4] = round(z * 32767.0);
+	output[(maxcomp + 0) % 4] = round(w * 32767.0);
+}
+```
+
+## Filter 3: exponential
+
+Exponential filter allows to encode floating point values with a range close to the full range of a 32-bit floating point value, but allows more control over the exponent/mantissa to trade quality for precision, and has a bit structure that is more optimally aligned to the byte boundary to facilitate better compression.
+
+This filter is only valid if `byteStride` is a multiple of 4.
+
+The input to the filter is a sequence of 32-bit little endian integers, with the most significant 8 bits specifying a (signed) exponent value, and the remaining 24 bits specifying a (signed) mantissa value. The integers are stored in two-complement format.
+
+The result of the filter is 2^e * m:
+
+```
+float32_t decode(int32_t input) {
+	int32_t e = input >> 24;
+	int32_t m = (input << 8) >> 8;
+	return pow(2.0, e) * m;
+}
+```
+
+The valid range of `e` is [-100, +100], which facilitates performant implementations. Decoding out of range values results in unspecified behavior, and encoders are expected to clamp `e` to the valid range.
+
+## Filter 4: color
+
+Color filter allows to encode color data using YCoCg color space transformation, which results in better compression for typical color data by exploiting correlation between color channels.
+
+This filter is only valid if `byteStride` is 4 or 8. When `byteStride` is 4, then the input and output of this filter are four 8-bit components, and when `byteStride` is 8, the input and output of this filter are four 16-bit components.
+
+The input to the filter is four 8-bit or 16-bit components, where the first component stores the Y (luma) value as a K-bit unsigned integer, the second and third components store Co/Cg (chrominance) values as K-bit signed integers, and the fourth component stores the alpha value as a K-1-bit unsigned integer with the bit K set to 1. 1 <= K <= 16, signed integers are stored in two's complement format.
+
+The transformation uses YCoCg encoding; reconstruction of RGB values can be performed in integer space or in floating point space, depending on the implementation. The encoder must guarantee that RGB can be reconstructed using K-bit integer math without overflow or underflow.
+
+The alpha component uses K-1 bits for the alpha value with the high bit set to 1; note that K can be smaller than the bit depth. This allows decoder to recover K and decode the color and alpha values correctly.
```

Comment:
What happens if the forth component is zero?

Reply:
This is invalid as written because `K >= 1`, and bit `K` must be set to `1` in the fourth component, so the fourth component can't be smaller than `1 << K`. I could add that as a clarification?